VirtoCommerce.CoreModule.Web/BackgroundJobs/IndexingJobs.cs

```diff
@@ -116,21 +117,20 @@ public async Task IndexChangesJob(string documentType, IJobCancellationToken can
                 try
                 {
                     // Create different notification for each option (document type)
-                    var executed = true;
+                    var success = true;
 
                     foreach (var options in o)
                     {
-                        executed = executed && await RunIndexJobAsync(null, null, true, new[] { options }, IndexChangesAsync, cancellationToken);
+                        success = success && await RunIndexJobAsync(null, null, true, new[] { options }, IndexChangesAsync, cancellationToken);
                     }
 
-                    if (executed)
-                    {
-                        // Indexation manager might re-use the jobs to scale out indexation.
-                        // Wait for all indexing jobs to complete, before telling interceptors we're ready.
-                        // This method is running as a job as well, so skip this job.
-                        // Scale out background indexation jobs are scheduled as low.
-                        await WaitForIndexationJobsToBeReadyAsync(JobPriority.Low, x => x.Method != _manualIndexAllJobMethod && x.Method != _indexChangesJobMethod);
-                    }
+                    // Indexation manager might re-use the jobs to scale out indexation.
+                    // Wait for all indexing jobs to complete, before telling interceptors we're ready.
+                    // This method is running as a job as well, so skip this job.
+                    // Scale out background indexation jobs are scheduled as low.
+                    await WaitForIndexationJobsToBeReadyAsync(JobPriority.Low, x => x.Method != _manualIndexAllJobMethod && x.Method != _indexChangesJobMethod);
```

Comment:
If we have result parameter success for RunIndexJobAsync method, i think we should leave the condition 
if (success)
{
 …
}
I see that in current case RunIndexJobAsync  could only succeed or fail with exception, but i do not think we should depend on that method inner logic in places where we use it.

Reply:
We should wait for all spawned jobs to complete whether success is true or not. So that the end of this task is really the end of the indexation. Is potentially important for the indexing interceptors.